### ISSUE TITLE
Add `--isoAnnotLite` flag to `sqanti3_RulesFilter.py`?

### ISSUE THREAD

**Issue by joelnitta** [ConesaLab/SQANTI3#30]
Hi! I am using SQANTI3 (v 1.3, current hash: 2190647b9c30dad3d28bc83eefbb983ea9f3c873) to prepare a gff3 file for tappAS.

Currently, I run `sqanti3_qc.py` first to generate `*_classification.txt`, then `sqanti3_RulesFilter.py` to filter for intrapriming. However, `sqanti3_RulesFilter.py` does not produce a gff3 file like `sqanti3_qc.py` would with the `--isoAnnotLite` flag. Could this option be added?

My work-around for now is to re-run `sqanti3_qc.py` again with the `--isoAnnotLite` flag using the `*filtered_lite.gtf` file output by `sqanti3_RulesFilter.py`. 

I also tried running `utilities/IsoAnnotLite_SQ3.py`, `utilities/IsoAnnotLite_SQ1.py`, and the [python script from the IsoAnnot Lite help page](https://isoannot.tappas.org/isoannot-lite/), but none of the them worked:

```
(SQANTI3.env) [joelnitta@juno sqanti3]$ python $BINDIR/SQANTI3/utilities/IsoAnnotLite_SQ3.py \
>   d_magna_classification.filtered_lite.gtf \
>   d_magna_classification.filtered_lite_classification.txt \
>   d_magna_classification.filtered_lite_junctions.txt


Running IsoAnnot Lite 1.5...
Traceback (most recent call last):
  File "/home/joelnitta/kato_daphnia/bin/SQANTI3/utilities/IsoAnnotLite_SQ3.py", line 1443, in <module>
    main()
  File "/home/joelnitta/kato_daphnia/bin/SQANTI3/utilities/IsoAnnotLite_SQ3.py", line 1320, in main
    run(args)
  File "/home/joelnitta/kato_daphnia/bin/SQANTI3/utilities/IsoAnnotLite_SQ3.py", line 1411, in run
    filename = os.path.join(args.dir , args.output+"_tappAS_annot_from_SQANTI3.gff3")
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```

```
(SQANTI3.env) [joelnitta@juno sqanti3]$ python $BINDIR/SQANTI3/utilities/IsoAnnotLite_SQ1.py \
>   d_magna_classification.filtered_lite.gtf \
>   d_magna_classification.filtered_lite_classification.txt \
>   d_magna_classification.filtered_lite_junctions.txt


Running IsoAnnot Lite 1.5...

Reading SQANTI 3 Files and creating an auxiliar GFF...
File classification does not have the correct structure. The column "ORF_length" is not in the possition 28 in the classification file. We have found the column "FSM_class".
```

```
(SQANTI3.env) [joelnitta@juno sqanti3]$ python $BINDIR/IsoAnnotLite/IsoAnnotLite1.5.py \
>   d_magna_classification.filtered_lite.gtf \
>   d_magna_classification.filtered_lite_classification.txt \
>   d_magna_classification.filtered_lite_junctions.txt


Running IsoAnnot Lite 1.5...

Reading SQANTI 3 Files and creating an auxiliar GFF...
File classification does not have the correct structure. The column "ORF_length" is not in the possition 28 in the classification file. We have found the column "FSM_class".
```

**Comment by aarzalluz** [comment]
Hi,

First of all, I apologize for the long time that it has taken for our team to get back to you.

Even though I am not the maintainer or developer of SQANTI3, I've been checking our repos for problems related to IsoAnnotLite, and have just seen that this issue is still open. I wanted to let you know [new version of IsoAnnotLite](https://isoannot.tappas.org/isoannot-lite/) is now available in our website. 

Feel free to try it out and see if your problems are solved. I believe that improvements made to the programme should be enough for you to be able to work with the `*filtered_lite.gtf` file... If not, please do let us know so that we can -finally- troubleshoot the problem.

I remember we've been in touch about tappAS before, so hopefully you'll be able to use the tool properly after this.

Best,

Ángeles

**Comment by aarzalluz** [comment]
Hi @joelnitta,

A user just reported a similar problem to the one you had before, but in this case with the newer IsoAnnotLite 2.0 (see [this issue](https://github.com/ConesaLab/tappAS/issues/16) in the tappAS repository where `sqanti3_RulesFilter.py` outputs are used as inputs to IsoAnnotLite). We'll work on it and get back to you.

I'm going to include my colleagues @FJPardoPalacios @Magdoll and @psalguerog in the issue so that they are aware on the problem and can get to it ASAP.

Best,

Ángeles

**Comment by joelnitta** [comment]
Thanks @aarzalluz for your response! I tried IsoAnnotLite 2.0 and it does seem to work now:

```
(SQANTI3.env) [joelnitta@juno temp]$ python $BINDIR/IsoAnnotLite_v2.0_SQ3.py \
>   ../d_magna_classification.filtered_lite.gtf \
>   ../d_magna_classification.filtered_lite_classification.txt \
>   ../d_magna_classification.filtered_lite_junctions.txt


Running IsoAnnot Lite 2.0...

Reading SQANTI 3 Files and creating an auxiliar GFF...
Adding extra information to relative columns...
Reading GFF3 to sort it correctly...
Generating final GFF3...
Time used to generate new GFF3: 3.03 seconds.

Exportation complete.
Your GFF3 result is: 'tappAS_annotation_from_SQANTI3.gff3'
```

However, although this is a work-around solution, I think it would still be more useful if `sqanti3_RulesFilter.py` had an `--isoAnnotLite` flag similar to `sqanti3_qc.py` as mentioned in the original post.

**Comment by aarzalluz** [comment]
Hi @joelnitta,

Thanks for the feedback! I'm glad it works now, even if it's through re-running IsoAnnotLite 2.0 with the filtered SQANTI3 files.

I believe adding the `--isoAnnotLite` flag to `sqanti3_RulesFilter.py` is a very good suggestion, and will surely be incorporated as part of SQANTI3 maintenance/development. I will leave the issue open for now so that the developers can keep track of the pending changes.

Best,


Ángeles

**Comment by aarzalluz** [comment]
**Fixed in version 5.0:** `sqanti3_RulesFilter.py` is now deprecated and has been replaced by `sqanti3_filter.py` (wrapper for both machine learning and rules filters). When supplied, the wrapper filters the `sqanti3_qc.py output` files using the filter result (rules or ML).
